test/integration/switch_locale_test.rb

```diff
@@ -0,0 +1,35 @@
+require "test_helper"
+
+class SwitchLocaleTest < ActionDispatch::IntegrationTest
+  test "I18n.locale will depends on the locale of urls" do
+    [:en, :"zh-TW"].each do |locale|
+      get root_path(locale:)
+      assert I18n.locale.to_s == locale.to_s
+    end
+  end
+
+  test "Visit urls with an unavailable locale will return 404" do
+    assert_raises ActionController::RoutingError do
+      get root_path(locale: "Klingon")
+    end
+  end
+
+  test "links to all locales will show" do
+    current_locale = :es
```

Comment:
You may have issues with this in the future. it is better to use `I18n.with_locale` when testing locale switches

From https://guides.rubyonrails.org/i18n.html

```
2.2 Managing the Locale across Requests

A localized application will likely need to provide support for multiple locales. To accomplish this, the locale should be set at the beginning of each request so that all strings are translated using the desired locale during the lifetime of that request.

The default locale is used for all translations unless I18n.locale= or I18n.with_locale is used.

I18n.locale can leak into subsequent requests served by the same thread/process if it is not consistently set in every controller. For example executing I18n.locale = :es in one POST requests will have effects for all later requests to controllers that don't set the locale, but only in that particular thread/process. For that reason, instead of I18n.locale = you can use I18n.with_locale which does not have this leak issue.
```

That can explain why an unrelated test is failing on CI

Reply:
@gmmcal @joemasilotti 
I have pushed [new commits](https://github.com/joemasilotti/railsdevs.com/pull/256#commits-pushed-5087736) to use `I18n.with_locale` instead.
Can you take a look?

Reply:
@hungmi these changes on #256 lgtm

Reply:
Thanks for the updates! I cherry-picked your commits onto this branch.